pkg/agent/manager/manager.go

```diff
@@ -159,22 +161,31 @@ func (m *manager) Initialize(ctx context.Context) error {
 	m.storeSVID(m.svid.State().SVID, m.svid.State().Reattestable)
 	m.storeBundle(m.cache.Bundle())
 
-	synchronizeBackoffMaxInterval := synchronizeMaxIntervalMultiple * m.c.SyncInterval
-	if synchronizeBackoffMaxInterval > synchronizeMaxInterval {
-		synchronizeBackoffMaxInterval = synchronizeMaxInterval
-	}
+	// upper limit of backoff is 8 mins
+	synchronizeBackoffMaxInterval := min(synchronizeMaxInterval, synchronizeMaxIntervalMultiple*m.c.SyncInterval)
+
 	m.synchronizeBackoff = backoff.NewBackoff(m.clk, m.c.SyncInterval, backoff.WithMaxInterval(synchronizeBackoffMaxInterval))
 	m.svidSyncBackoff = backoff.NewBackoff(m.clk, cache.SVIDSyncInterval, backoff.WithMaxInterval(maxSVIDSyncInterval))
 	m.csrSizeLimitedBackoff = backoff.NewSizeLimitedBackOff(limits.SignLimitPerIP)
 
-	err := m.synchronize(ctx)
-	if nodeutil.ShouldAgentReattest(err) {
+	err := m.handleSyncError(m.synchronize(ctx))
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (m *manager) handleSyncError(err error) error {
```

Comment:
I don't really see a benefit of making this a dedicated method since it's only used to handle an error from one specific function call. Would you mind moving the error handling back to where it was before?